dynamicio/config.py

```diff
@@ -104,6 +104,53 @@ def dyn_str_constructor(self, node: yaml.nodes.ScalarNode) -> str:
         return value
 
 
+class SafeDynamicSchemaLoader(yaml.SafeLoader):
+    """Implements a dynamic yaml loader that parses yaml files and replaces strings that map to [[ DYNAMIC_VAR ]].
+
+    Dynamic variables defined in a provided module object.
+    """
+
+    module = None
+    dynamic_data_matcher = re.compile(r"(.*)(\[\[\s*(\S+)\s*]])(.*)")
+
+    @classmethod
+    def with_module(cls, module: ModuleType):
+        """Creates a dynamic subclass of SafeDynamicLoader with the `data_module` attribute set to `module`.
+
+        Args:
+            module: A global vars module with all the dynamic values defined in it.
+
+        Returns:
+            type
+        """
+        return type(f"{cls.__name__}_{module.__name__}", (cls,), {"module": module})
```

Comment:
This line seems interesting, i've not come across this type of syntax, would you mind explaining what is happening here?

Reply:
What happens if you want to use something like: `"[[ VALUE_1 ]]/[[ VALUE_2 ]]"'`

Reply:
Hey Christiano. You might be familiar with the other form of type, which you may have used as `cls = type(some_instance)`, which sets `cls` to the class reference of a`some_instance`'s type.

This other form you see here is `type(cls_name: str, bases: Tuple[Type], members: Mapping[str, Any])` and it returns a new class! So, here, this is creating a new class named `{class name}_{module name}` which is a subclass of the current class and has one attribute `module` set to the object to which `module` points to.

This trick is used to bind a module holding the values that must be replaced to resolve the config to an actual config class.